### PR TITLE
Add OAuth2 support

### PR DESCRIPTION
**Pull request by nb-ohad**
- Discover OAuth2 endpoints using the .wellknown auth endpint and relay them to noobaa-core via env vars
- Relay noobaa service account to noobaa-core pod via env variable
- CLI: Add missing service account annotaion and cluster role binding (to system:auth-delegator)

### CODE REVIEW

**Comment by guymguym** [comment]
@nb-ohad make test fails in CI:
```
# github.com/noobaa/noobaa-operator/v2/pkg/system
pkg/system/phase2_creating.go:33:18: Errorf call has arguments but no formatting directives
```

File: pkg/system/phase2_creating.go [hunk 1]
```diff
@@ -27,6 +27,13 @@ func (r *Reconciler) ReconcilePhaseCreating() error {
 		"noobaa operator started phase 2/4 - \"Creating\"",
 	)
 
+	// A failiure to discover OAuth endpoints should not fail the entire reconcile phase.
+	oAuthEndpoints, err := util.DiscoverOAuthEndpoints()
+	if err != nil {
+		r.Logger.Errorf("Discovery of OAuth endpoints failed, got:", err)
```
Comment (guymguym) [inline pkg/system/phase2_creating.go]: This is the line of the CI failure - Errorf requires `%s` in the message for the error arg

Comment (nb-ohad) [reply pkg/system/phase2_creating.go]: Not specifically the whole operation is optional. Many env will not support the .well-known endpoint: Vennila k8s, OpenShift prior to 4

File: pkg/util/util.go [hunk 2]
```diff
@@ -721,3 +733,30 @@ func PrintThisNoteWhenFinishedApplyingAndStartWaitLoop() {
 	log.Printf("  - From now on, it only loops and reads the status, to monitor the operator work.")
 	log.Printf("  - You may Ctrl-C at any time to stop the loop and watch it manually.")
 }
+
+func DiscoverOAuthEndpoints() (*OAuth2Endpoints, error) {
```
Comment (guymguym) [inline pkg/util/util.go]: I see that oauth-proxy has the same logic here:
https://github.com/openshift/oauth-proxy/blob/fbd062bbcb23538863a735f9823ac3fd6f47429e/providers/openshift/provider.go#L543-L567

Do you think we can/should call that code instead?

File: pkg/util/util.go [hunk 3]
```diff
@@ -721,3 +733,30 @@ func PrintThisNoteWhenFinishedApplyingAndStartWaitLoop() {
...
+	client := http.Client{
+		Timeout: 120 * time.Second,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
```
[the remark this answers is not in the document]
Comment (nb-ohad) [reply pkg/util/util.go]: In order to remove the InsecureSkipVerify option, the operator needs to understand that service-ca.crt on the operator pod is a valid root ca. Which is not currently the case.
We can find a way to do this and add it in a future change

File: pkg/system/phase2_creating.go (hunk 1 above)
Comment (guymguym) [reply pkg/system/phase2_creating.go]: Ah that's right.
So can you make sure this http call doesn't hang when running on vanilla kube?

File: pkg/util/util.go (hunk 2 above)
Comment (nb-ohad) [reply pkg/util/util.go]: We are not using OAuth proxy or its codebase

File: pkg/util/util.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (nb-ohad) [reply pkg/util/util.go]: I will open an Issue

**Pull request merged**